src/Momento.Sdk/Config/Middleware/IMiddleware.cs

```diff
@@ -36,7 +36,7 @@ Func<Metadata> GetTrailers
 /// </summary>
 public interface IMiddleware
 {
-    public ILoggerFactory LoggerFactory { get; }
+    public ILoggerFactory? LoggerFactory { get; }
 
     public IMiddleware WithLoggerFactory(ILoggerFactory loggerFactory);
```

Comment:
Why do these live on IMiddleware instead of being concerns of the implementation? Conceptually, a LoggerFactory getter seems orthogonal to the job of an IMiddleware, as advertised by its interface.

Reply:
The configuration object is where a user defines their logging configuration.

The user may also pass in a list of Middlewares and I don't want or need them to have to also pass in the logging config to each and every one of those Middlewares.

Therefore the Configuration object needs to be able to loop over the middlewares and configure their logging, if the user didn't explicitly add a logging config to the middleware.

Therefore the IMiddleware interface must expose a way for the Configuration to get/set their logging configuration.

Reply:
> The Configuration object is where a user defines their logging configuration.

Sure, so shouldn't middleware implementations which consume Configuration either take in Configuration or their relevant explicit Configuration dependencies in their constructors?

if the user didn't add logging, then why does their middleware need logging configuration? For middleware-framework logging it seems like you'd want a middleware-framework logger instance, no? And for middleware-implementation logging, if you don't log then why should there be an interface dependency?

I think this pattern (migrating from `ILoggerFactory` to `ILoggerFactory?`) is mixing setter-injection and constructor-injection. It's unclear to me how setter-injection is made appropriate or necessary by this pr. Maybe we should chat or zoom separately

Reply:
Discussed this with Kenny offline.  I agree there's some code smell on this and we talked through a few ideas to experiment with.  The scope of those changes is bigger than appropriate for this PR so I am going to try to find some time to look at that separately later today.